client/client.go

```diff
@@ -326,11 +335,11 @@ func decodeInto(reader io.Reader, v interface{}) error {
 	return nil
 }
 
-// doSync performs a request to the given path using the specified HTTP method.
+// DoSync performs a request to the given path using the specified HTTP method.
 // It expects a "sync" response from the API and on success decodes the JSON
 // response payload into the given value using the "UseNumber" json decoding
 // which produces json.Numbers instead of float64 types for numbers.
-func (client *Client) doSync(method, path string, query url.Values, headers map[string]string, body io.Reader, v interface{}) (*ResultInfo, error) {
+func (client *Client) DoSync(method, path string, query url.Values, headers map[string]string, body io.Reader, v interface{}) (*ResultInfo, error) {
```

Comment:
Why does `DoSync` need to be exported?

Reply:
Please re-review this PR since it makes much more sense now.
Answering to your concrete question, applications extending the Pebble client will definitely want these utility functions to be available in order to communicate with the Pebble daemon.